torchdynamo/variables/torch.py

```diff
@@ -158,6 +159,22 @@ def call_function(
             and args[0].size is not None
         ):
             return ConstantVariable(product(args[0].size), **options)
+        elif (
+            self.value
+            in (
+                torch._shape_as_tensor,
+                torch.onnx.operators.shape_as_tensor,
+            )
+            and config.rewrite_unsafe_ops
+        ):
+            # We don't want to get rid of `_shape_as_tensor` during export to ONNX phase
+            if (not torch.jit.is_scripting()) and (not torch.onnx.is_in_onnx_export()):
```

Comment:
I don't think these can ever be true when dynamo is doing analysis.

Reply:
I see thanks :)